tasks/internal_reconcile_report_orphan/adapters/database/postgres.py

```diff
@@ -0,0 +1,108 @@
+import dataclasses
+import logging
+
+from orca_shared.database import shared_db
+from sqlalchemy import text
+from sqlalchemy.future import Engine
+
+from entities.orphan import OrphanRecordFilter, OrphanRecordPage, OrphanRecord
+from use_cases.adapter_interfaces.database import DatabaseInterface
+
+
+@dataclasses.dataclass  # todo: Replace the loosely-defined dictionary in orca_shared.database with this.
```

Comment:
Given the amount of code for pulling a secret and parsing into db_connect_info, I feel it still makes sense to use the shared lib. Recommend we use a proper data class as a return value instead of a dict so we can keep the returned data-type architecture-specific.